tensornetwork/network_components.py

```diff
@@ -1074,25 +1069,25 @@ def node2(self) -> Optional[BaseNode]:
           'Edge has been disabled, accessing node2 is no longer possible')
     if self._is_dangling:
       return None
-    if self._node2 is None:
+    if self._nodes[1] is None:
       raise ValueError("node2 for edge '{}' no longer exists.".format(self))
-    return self._node2
+    return self._nodes[1]
 
   @node1.setter
   def node1(self, node: BaseNode) -> None:
     if self.is_disabled:
       raise ValueError(
           'Edge has been disabled, setting node1 is no longer possible')
     # pylint: disable=attribute-defined-outside-init
-    self._node1 = node
+    self._nodes[0] = node
 
   @node2.setter
   def node2(self, node: Optional[BaseNode]) -> None:
```

Comment:
To support arbitrary number of nodes, seems like it would be cleaner to have a single getter and a single setter that takes the node number as an argument.

Reply:
Good point,for now this is just to allow the existing code to work. Ultimately, I plan on completely removing the `axis1, axis2, node1, node2` entirely. This will take many PRs as I basically have to modify all of the existing `Edge` logic.

And that getter and setter that you mentioned will basically just be what the `update_axis` method currently is.